src/ui/Search/frmsearchreplace.cpp

```diff
@@ -535,15 +535,16 @@ void frmSearchReplace::on_searchStringEdited(const QString &/*text*/)
 
     if (s.Search.getSearchAsIType()) {
         if (ui->actionFind->isChecked()) {
-            Editor *editor = currentEditor();
-
-            QList<Editor::Selection> selections = editor->getSelections();
-            if (selections.length() > 0) {
-                editor->setCursorPosition(
-                            std::min(selections[0].from, selections[0].to));
-            }
-
-            findFromUI(true);
+            currentEditor()->getSelections([&](const QList<Editor::Selection> selections) {
+                Editor *editor = currentEditor();
```

Comment:
Can you move the assignment to `*editor` outside the callback and pass it in as a closure?